FinSearchUnified/Bundle/SearchBundleFindologic/FacetHandler/CategoryFacetHandler.php

```diff
@@ -0,0 +1,166 @@
+use Shopware\Bundle\SearchBundle\FacetResult\TreeFacetResult;
+use Shopware\Bundle\SearchBundle\FacetResult\TreeItem;
+use SimpleXMLElement;
+
+class CategoryFacetHandler implements PartialFacetHandlerInterface
+{
+    /**
+     * @param FacetInterface $facet
+     * @param Criteria $criteria
+     * @param SimpleXMLElement $filter
+     *
+     * @return TreeFacetResult
+     */
+    public function generatePartialFacet(FacetInterface $facet, Criteria $criteria, SimpleXMLElement $filter)
+    {
+        $categories = $this->getActives($facet, $criteria);
+        $categories = $this->parseCategories($filter->items->item, $categories);
+
+        return new TreeFacetResult(
+            $facet->getField(),
+            $facet->getFormFieldName(),
+            $criteria->hasCondition($facet->getName()),
+            $facet->getLabel(),
+            $this->getTreeItems($categories)
+        );
+    }
+
+    /**
+     * @param SimpleXMLElement $filter
+     *
+     * @return bool
+     */
+    public function supportsFilter(SimpleXMLElement $filter)
+    {
+        return (string)$filter->name === 'cat';
+    }
+
+    /**
+     * Parses the active values and returns an array
+     *
+     * @param FacetInterface $facet
+     * @param Criteria $criteria
+     *
+     * @return array
+     */
+    private function getActives(FacetInterface $facet, Criteria $criteria)
+    {
+        $actives = [];
+
+        $condition = $criteria->getCondition($facet->getName());
+
+        if ($condition !== null) {
+            $actives = $condition->getValue();
+        }
+
+        if (!is_array($actives)) {
+            $actives = [$actives];
+        }
+
+        $categories = $this->prepareCategoryTree($actives);
+
+        return $categories;
+    }
+
+    /**
+     * Helper method to recursively create category tree
+     *
+     * @param $actives
+     * @param array $categories
+     *
+     * @return mixed
+     */
+    private function prepareCategoryTree($actives, array $categories = [])
+    {
+        foreach ($actives as $active) {
+            list($parent, $child) = explode('_', $active);
+
+            $categories[$parent] = [
+                'active' => $child ? false : true,
+                'children' => $child ? self::prepareCategoryTree([$child], $categories) : []
+            ];
+        }
+
+        return $categories;
+    }
+
+    /**
+     * Parse the filter items and build an array structure
+     *
+     * @param SimpleXMLElement $filterItems
+     * @param array $actives
+     *
+     * @return array
+     */
+    private function parseCategories(SimpleXMLElement $filterItems, array $actives)
+    {
+        foreach ($filterItems as $filterItem) {
+            $name = (string)$filterItem->name;
+            $frequency = (int)$filterItem->frequency;
+
+            // Only add the items of $filterItems that are not already present in $actives
+            $index = array_search($name, $actives);
```

Comment:
This will not work since `$actives` is a multi dimensional array. You will have to search the array recursively by not only considering the current filter item, but also the parents of it.